src/RequestHandler/outofprocess/serverprocess.cxx

```diff
@@ -1006,29 +1069,11 @@ SERVER_PROCESS::StartProcess(
             pHashTable = NULL;
         }
 
-        if (FAILED(hr))
-        {
-            if (strEventMsg.IsEmpty())
-            {
-                strEventMsg.SafeSnwprintf(
-                    ASPNETCORE_EVENT_PROCESS_START_POSTCREATE_ERROR_MSG,
-                    m_struAppFullPath.QueryStr(),
-                    m_struPhysicalPath.QueryStr(),
-                    m_struCommandLine.QueryStr(),
-                    hr,
-                    dwRetryCount);
-            }
+        CleanUp();
```

Comment:
Do we only want to cleanup if we successfully log an error message? Also, cleanup will remove all child process rather than just the one that failed. I think we only want to remove the child process that failed to start.

Reply:
Cleanup will reset the handles and callbacks. We want to do cleanup so that we can try restart the process again. we want to stop all child processes associated with the process created by the CreateProcess call (scenario that user put script as commandline which may start a set of processes)